src/com/lftechnology/java/training/niraj/ems/controllers/EmployeeController.java

```diff
@@ -0,0 +1,127 @@
+            }
+            RouteServices.routeAfterLogin(loggedInUser);
+        } catch (InputMismatchException | SQLException se) {
+            LOGGER.severe("There was some error. Please try again later.");
+            LOGGER.log(Level.SEVERE, "Exception:{0}", se);
+
+        }
+
+    }
+
+    /**
+     * Edit profile of the logged in user
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param employee
+     *            {@link Employee}
+     * 
+     */
+    public void editProfile(Employee employee) {
+        try (Scanner scanner = new Scanner(System.in)) {
+            Console console = System.console();
+            LOGGER.info(Constants.EDIT_MENU);
+            userService.editUserProfile(employee, scanner, console);
+            LOGGER.info(Constants.EDIT_SUCCESS);
+            RouteServices.routeAfterLogin(employee);
+        } catch (InputMismatchException | SQLException se) {
+            LOGGER.log(Level.SEVERE, "Exception: {0}", se);
+        }
+
+    }
+
+    /**
+     * Search an employee by username
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param employee
+     *            {@link Employee}
+     */
+    public void search(Employee employee) {
+        try (Scanner scanner = new Scanner(System.in)) {
+            Console console = System.console();
+            String searchResult = userService.searchUser(scanner, console);
+            LOGGER.info(searchResult);
+            RouteServices.routeAfterLogin(employee);
+        } catch (InputMismatchException | SQLException se) {
+            LOGGER.severe("Failed to retrieve the search results. please try again.");
+            LOGGER.log(Level.SEVERE, "Exception:{0}", se);
+        }
+
+    }
+
+    /**
+     * Lists all the active users
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     * @param employee
+     */
+    public void listUsers(Employee employee) {
+        try (Scanner scanner = new Scanner(System.in)) {
+            Console console = System.console();
+            String userList = userService.getUsersList();
+            LOGGER.info(userList);
+            userService.TerminateDeleteUser(scanner, console, employee);
+            RouteServices.routeAfterLogin(employee);
+
+        } catch (InputMismatchException | SQLException se) {
+            LOGGER.severe("Failed to retrieve the user lists. please try again.");
+            LOGGER.log(Level.SEVERE, "Exception:{0}", se);
+            RouteServices.routeAfterLogin(employee);
+        } catch (CustomException ce) {
+            LOGGER.log(Level.SEVERE, "Exception:{0}", ce);
+        }
+    }
+
+    /**
+     * Logs the user out of the system
+     * 
+     * @author Niraj Rajbhandari <nirajrajbhandari@lftechnology.com>
+     */
+    public void logout() {
+        LOGGER.info(Constants.GOODBYE_MSG);
+        System.exit(0);
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Remove this call to "exit" or ensure it is really required. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1147)